src/ophyd_async/epics/areadetector/controllers/ad_controller.py

```diff
@@ -0,0 +1,42 @@
+import asyncio
+from typing import Optional
+
+from ophyd_async.core import (
+    DEFAULT_TIMEOUT,
+    AsyncStatus,
+    DetectorControl,
+    DetectorTrigger,
+    set_and_wait_for_value,
+    wait_for_value,
+)
+
+from ..drivers.ad_driver import ADDriver, ImageMode
+
+
+class ADController(DetectorControl):
+    def __init__(self, driver: ADDriver) -> None:
+        self.driver = driver
+
+    def get_deadtime(self, exposure: float) -> float:
+        return 0.002
+
+    async def arm(
+        self,
+        trigger: DetectorTrigger = DetectorTrigger.internal,
```

Comment:
Should assert `trigger == DetectorTrigger.internal` as we don't handle anything else